### PR TITLE
Support for Julia

### PR DESCRIPTION
**Pull request by stevenheidel**
This is what it looks like:

```julia
"""
A single request object according to the JSONRPC standard.
"""
Base.@kwdef struct RPCRequest
    "The JSONRPC version."
    jsonrpc::Union{String, Nothing} = "2.0"

    "The RPC function name."
    method::String

    "The RPC function arguments."
    params::Any

    "RPC request id (used to verify that request and response belong together)."
    id::String
end
```

### CODE REVIEW

File: src/rpcq-julia.lisp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    ((cons (eql :map))
+     (assert (string= (symbol-name (third field-type)) "->")
+             (field-type)
+             "Bad mapping spec.")
+     (format nil
+             "Dict{~A, ~A}"
+             (julia-type (second field-type))
+             (julia-type (fourth field-type))))))
+
+(defun julia-optional-type (field-type required)
```
Comment (ecpeterson) [inline src/rpcq-julia.lisp]: but where's `julia-maybe-optional-typing-type`

**Comment by notmgsk** [comment]
@stevenheidel Would you mind if I forked this into a proper `RPCQ.jl` repo? I think Julia likes dedicated git repos for package management + I have some stuff I want to commit.

**Comment by notmgsk** [comment]
I'm gonna do it anyway, just try and stop me.

**Comment by stevenheidel** [comment]
sgtm 👍

**Pull request closed without merging**